### ISSUE TITLE
Add a pre-commit hook to check for syntax errors

### ISSUE THREAD

**Issue by bobsany16** [VickiLanger/LGBTQ-of-the-day-bot#84]
**Is your feature request related to a problem? Please describe.**
Lots of times, entries to data files contain syntax errors like commas, periods, etc., 

**Describe the solution you'd like**
Have an automated pre-commit hook that will check for these errors. If there are, the hook won't allow a push => prompts contributors to check their changes. 
This -> makes things easier for maintainer/organizer -> good practice of checking code for errors for everyone. 

**Additional context**

---
*Don't forget to follow the bot on [Twitter](https://www.twitter.com/LGBTQotd) to see the hard work you put in.*

**Comment by VickiLanger** [comment]
Oh! I had no clue that was something that could be done.

**Comment by bobsany16** [comment]
Yup! I've seen a couple projects implement multiple sorts of pre-commit hooks but unfortunately, I have 0 experience with this. Would be a blast to work on though 😅. Unless someone can chime in for this, I'll attempt.

**Comment by bobsany16** [comment]
I found this, I might try [this](https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/), if that's cool with you

**Comment by VickiLanger** [comment]
Neat. I thought it was going to be some [github actions](https://github.com/VickiLanger/Queer-of-the-day-bot/actions/new) type of thing. You're welcome to try it.

**Comment by VickiLanger** [comment]
If you haven't put too much work into the hooks. I think this [python linter bot](https://github.com/marketplace/actions/python-linter) may do what we need

**Comment by bobsany16** [comment]
oh awesome! I was just still doing some researches on the hooks but unfortunately wasn't doing too good 😅 and the python linter bot does make things easier. I assume it will check for errors and then if it doesn't pass, we could request changes in the PR, right ? If so, I think this is more feasible. Thanks for letting me know!

**Comment by VickiLanger** [comment]
Um, I didn't look too hard yet. Just ran across it looking for something else

**Comment by bobsany16** [comment]
alright, I'll take a look and let you know!

**Comment by bobsany16** [comment]
Do you mind sharing the repo that uses this ? I just wanted to see their .yml file configurations.

**Comment by VickiLanger** [comment]
I found it in the GitHub actions page. I don't know of anyone specific who uses it. Though I'm sure many do use it

**Comment by lhfaguiar** [comment]
have you guys figured something out? I'm looking for a way of making it stop failing at the same point every PR

**Comment by VickiLanger** [comment]
> have you guys figured something out? I'm looking for a way of making it stop failing at the same point every PR

Personally, I haven't worked on it. Not sure if @bobsany16 has.
